compiler/rustc_hir_typeck/src/generator_interior/mod.rs

```diff
@@ -643,15 +643,19 @@ fn check_must_not_suspend_ty<'tcx>(
         }
         ty::Array(ty, len) => {
             let descr_pre = &format!("{}array{} of ", data.descr_pre, plural_suffix);
+            let target_usize = len.try_eval_target_usize(fcx.tcx, fcx.param_env).unwrap_or(0);
+            let plural_len = if target_usize == usize::MAX {
+                usize::MAX
+            } else {
+                computed_len + 1
+            };
```

Comment:
This is essentially [`usize::saturating_add`](https://doc.rust-lang.org/std/primitive.usize.html#method.saturating_add):
```suggestion
            let target_usize = len.try_eval_target_usize(fcx.tcx, fcx.param_env).unwrap_or(0) as usize;
            let plural_len = target_usize.saturating_add(1);
```